### PR TITLE
feat(initial-repository): add user instruction and put the repository in its initial state

### PR DESCRIPTION
**Pull request by l-thienot**
proposition d'état initial + mini guide apprenant 

Co-authored-by: Florian Gomas

### CODE REVIEW

File: src/main/java/com/github/jtama/crazy/dop/SuitCard.java [hunk 1]
```diff
@@ -1,5 +1,23 @@
 package com.github.jtama.crazy.dop;
 
-public sealed interface SuitCard extends PlayingCard permits NumberSuitCard, RoyalSuitCard {
-    Color color();
+public final class SuitCard implements PlayingCard {
```
Comment (jtama) [inline src/main/java/com/github/jtama/crazy/dop/SuitCard.java]: We should implement equals, hashcode, and toString(), juste to show how much records make that easy.

File: src/main/java/com/github/jtama/crazy/dop/PrettyPrinter.java [hunk 2]
```diff
@@ -15,14 +15,29 @@ public static String toString(List<PlayingCard> cards) {
...
-            case NumberSuitCard suitCard when suitCard.index() == 1 -> "The first of %s(%s) is very weak".formatted(suitCard.color().name().toLowerCase(), suitCard.color().getSymbol());
-            case NumberSuitCard suitCard when suitCard.index() == 2 -> "The second of %s(%s) is very weak".formatted(suitCard.color().name().toLowerCase(), suitCard.color().getSymbol());
-            case NumberSuitCard suitCard when suitCard.index() == 3 -> "The third of %s(%s) is very weak".formatted(suitCard.color().name().toLowerCase(), suitCard.color().getSymbol());
-            case NumberSuitCard suitCard when suitCard.index() < 7 -> "The %sth of %s(%s) is still weak".formatted(suitCard.index(),suitCard.color().name().toLowerCase(), suitCard.color().getSymbol());
-            case NumberSuitCard(Color color, Integer index) -> "The %sth of %s(%s) may win you a hand".formatted(index, color.name().toLowerCase(), color.getSymbol());
-            case RoyalSuitCard(Color color, Face face) -> "The %s of %s(%s) is strong".formatted(face.displayName(), color.name().toLowerCase(), color.getSymbol());
-        };
+        if(playingCard instanceof SuitCard suitCard){
```
Comment (jtama) [inline src/main/java/com/github/jtama/crazy/dop/PrettyPrinter.java]: I think that's already pattern matching in the initial state. I'd rather make en initial state that is 100% java 8 compatible.

What do you think ?

**Review by jtama — CHANGES_REQUESTED** [review]

File: src/main/java/com/github/jtama/crazy/dop/PrettyPrinter.java (hunk 2 above)
Comment (yrekass) [reply src/main/java/com/github/jtama/crazy/dop/PrettyPrinter.java]: If I have correctly grasped the flow of the workshop, we should start by creating a branch called 'start' that utilizes classic Java 8. The initial step will involve replacing all records with traditional Java classes. Next, we will introduce the concept of records and demonstrate how they can enhance code clarity. Finally, in the PrettyPrinter class, we will revert the conditional logic to a standard if statement and apply pattern matching to reach the desired outcome.

Comment (jtama) [reply src/main/java/com/github/jtama/crazy/dop/PrettyPrinter.java]: That's exactly what's expected. And the start branch should be as ugly as possible.
And contain the documentation with eache expected steps

**Comment by jtama** [comment]
Super départ !

Tu squashs les commits et on merge.

**Review by jtama — APPROVED** [review]
Super

**Pull request merged**